src/renderer/components/+extensions/extensions.tsx

```diff
@@ -32,27 +60,264 @@ export class Extensions extends React.Component {
+      await fse.ensureDir(unpackingTempFolder);
+      await extractTar(tmpFile, { cwd: unpackingTempFolder });
+
+      // move contents to extensions folder
+      const unpackedFiles = await fse.readdir(unpackingTempFolder);
+      let unpackedRootFolder = unpackingTempFolder;
+      if (unpackedFiles.length === 1) {
+        // check if %extension.tgz was packed with single top folder,
+        // e.g. "npm pack %ext_name" downloads file with "package" root folder within tarball
+        unpackedRootFolder = path.join(unpackingTempFolder, unpackedFiles[0]);
+      }
+      await fse.ensureDir(extensionFolder);
+      await fse.move(unpackedRootFolder, extensionFolder, { overwrite: true });
+      Notifications.ok(
+        <p>Extension <b>{extName}</b> successfully installed!</p>
+      );
+    } catch (err) {
+      Notifications.error(
+        <p>Installing extension <b>{extName}</b> has failed: <em>{err}</em></p>
+      );
+    } finally {
+      // clean up
+      fse.remove(unpackingTempFolder).catch(Function);
+      fse.unlink(tmpFile).catch(Function);
+    }
+  }
+
   renderInfo() {
     return (
-      <div className="flex column gaps">
+      <div className="extensions-info flex column gaps">
         <h2>Lens Extension API</h2>
         <div>
           The Extensions API in Lens allows users to customize and enhance the Lens experience by creating their own menus or page content that is extended from the existing pages. Many of the core
           features of Lens are built as extensions and use the same Extension API.
         </div>
         <div>
-          Extensions loaded from:
-          <div className="extensions-path flex inline">
+          <SubTitle title="Extensions loaded from:"/>
+          <div className="extensions-path flex inline" onClick={() => shell.openPath(this.extensionsPath)}>
+            <Icon material="folder" tooltip={{ children: "Open folder", preferredPositions: "bottom" }}/>
             <code>{this.extensionsPath}</code>
-            <Icon
-              material="folder"
-              tooltip="Open folder"
-              onClick={() => shell.openPath(this.extensionsPath)}
-            />
           </div>
         </div>
-        <div>
-          Check out documentation to <a href="https://docs.k8slens.dev/" target="_blank">learn more</a>
+        <div className="install-extension flex column gaps">
+          <SubTitle title="Install extensions:"/>
+          <Input
+            showErrorsAsTooltip={true}
+            className="box grow"
+            theme="round-black"
+            iconLeft="link"
+            placeholder={`URL to packed extension (${this.supportedFormats.join(", ")})`}
+            validators={InputValidators.isUrl}
+            value={this.downloadUrl}
+            onChange={v => this.downloadUrl = v}
+            onSubmit={this.addExtensions}
+          />
+          <Button
+            primary
+            label="Add extensions"
```

Comment:
This should be i18n-ed

Reply:
Until we officially support i18n it is NOT (and I recommend to everyone don't spend time on it)